engine-comparison/dispatcher.sh

```diff
@@ -148,11 +148,102 @@ for FENGINE_CONFIG in $(find ${WORK}/fengine-configs/*); do
   done
 done
 
-# rm -rf $WORK/SEND*
+# TODO here: reset fengine-config env vars (?)
+############# DONE building
+rm -rf $WORK/send $WORK/build
+############# NOW wait for experiment results
+
+# Make a coverage build for a benchmark
+make_measurer () {
+  BENCHMARK=$1
+
+  BUILDING_DIR=$WORK/coverage-builds/${BENCHMARK}
+  mkdir -p $BUILDING_DIR
+  cd $BUILDING_DIR
+  FUZZING_ENGINE=coverage $WORK/FTS/${BENCHMARK}/build.sh
+}
+
+# Process a corpus, generate a human readable report, send the report to gsutil
+# Processing includes: decompress, use sancov.py, etc
+measure_coverage () {
+  FENGINE_CONFIG=$1
+  FENGINE_NAME=$(basename $FENGINE_CONFIG)
+  BENCHMARK=$2
+  CYCLE=$3
+
+  CORPUS_DIR=$WORK/measurement-folders/$BENCHMARK/corpus
+  SANCOV_DIR=$WORK/measurement-folders/$BENCHMARK/sancovs
+  REPORT_DIR=$WORK/measurement-folders/$BENCHMARK/reports
+  rm -fr $CORPUS_DIR $SANCOV_DIR
+  mkdir -p $CORPUS_DIR $SANCOV_DIR $REPORT_DIR
+
+  cd $CORPUS_DIR
+  # tar flags specify location to extract contents to (CORPUS_DIR)
+  tar -xvf $WORK/experiment-folders/${BENCHMARK}-${FENGINE_NAME}/corpus/corpus-archive-${CYCLE}.tar.gz -C $CORPUS_DIR --strip-components=1
+
+  # Generate individual sancovs
+  cd $SANCOV_DIR
+  for fl in $(find $CORPUS_DIR -type f); do
+    UBSAN_OPTIONS=coverage=1 $WORK/coverage-builds/${BENCHMARK}/${BENCHMARK}-coverage $fl
+  done
```

Comment:
Might be easier to do `UBSAN_OPTIONS=coverage=1 $WORK/coverage-builds/${BENCHMARK}/${BENCHMARK}-coverage $(find $CORPUS_DIR -type f)`

I think that would eliminate the merge step below.